src/lib.rs

```diff
@@ -27,16 +27,92 @@ extern crate memmap;
 extern crate object;
 extern crate owning_ref;
 extern crate fallible_iterator;
+#[macro_use]
+extern crate error_chain;
 
 use object::Object;
 use owning_ref::OwningHandle;
 use fallible_iterator::FallibleIterator;
 
+use std::fmt;
 use std::path;
+use std::error;
 use std::borrow::Cow;
 
-pub mod error;
-pub use error::*;
+/// An error occured while traversing the debug symbols in the provided executable.
+#[derive(Debug)]
+pub enum DebugInfoError {
```

Comment:
Any reason why this is a separate type from the `error` module's `Error` rather than just having one common error type for the whole library?